tf_agents/networks/multi_categorical_projection_network.py

```diff
@@ -0,0 +1,140 @@
+"""Project inputs to multiple categorical distribution object."""
+
+import gin
+import numpy as np
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+from tf_agents.networks import network
+from tf_agents.networks import utils
+from tf_agents.specs import tensor_spec
+from tf_agents.specs import distribution_spec
+
+class MultiCategoricalDistributionBlock(tfp.distributions.Blockwise):
+
+  def __init__(self, logits, categories_shape):
+    self.categories_shape = categories_shape
+    distribs = self._create_distrib(logits)
+    super().__init__(distributions = distribs)
+    self._parameters['logits'] = logits
+
+  def _create_distrib(self, logits):
+    logits_splitted = tf.split(logits, self.categories_shape, -1)
+    distribs = tf.nest.map_structure(
+                    lambda l: tfp.distributions.Categorical(logits = l),
+                    logits_splitted)
+    return distribs
+
+  def _mode(self):
+    return self._flatten_and_concat_event(
+      self._distribution.mode()
+      )
+
+@gin.configurable
+class MultiCategoricalProjectionNetwork(network.DistributionNetwork):
+  """Generates a set of tfp.distribution.Categorical by predicting logits"""
+  def __init__(self,
+            sample_spec,
+            logits_init_output_factor=0.1,
+            name='MultiCategoricalProjectionNetwork'):
+    """Creates an instance of MultiCategoricalProjectionNetwork
+
+    Args:
+      sample_spec: A collection of `tensor_spec.BoundedTensorSpec` detailing
+        the shape and dtypes of samples pulled from the output distribution.
+      logits_init_output_factor: Output factor for initializing kernal
```

Comment:
Not clear what does it mean output factor?
kernal -> kernel

Reply:
Fixed kernel spelling error. 

The descriptor for `logits_init_output_factor` was inherited from `CategoricalProjectionNetwork`, but I do agree it is not exactly clear what output factor means here.